sphinx/environment/__init__.py

```diff
@@ -676,19 +676,20 @@ def read_doc(self, docname, app=None):
 
         language = self.config.language or 'en'
         self.settings['language_code'] = language
-        self.settings['smart_quotes'] = True
-        if self.config.html_use_smartypants is not None:
-            warnings.warn("html_use_smartypants option is deprecated. Smart "
-                          "quotes are on by default; if you want to disable "
-                          "or customize them, use the smart_quotes option in "
-                          "docutils.conf.",
-                          RemovedInSphinx17Warning)
-            self.settings['smart_quotes'] = self.config.html_use_smartypants
-        for tag in normalize_language_tag(language):
-            if tag in smartchars.quotes:
-                break
-        else:
-            self.settings['smart_quotes'] = False
+        if 'smart_quotes' not in self.settings:
+            self.settings['smart_quotes'] = True
+            if self.config.html_use_smartypants is not None:
+                warnings.warn("html_use_smartypants option is deprecated. Smart "
+                              "quotes are on by default; if you want to disable "
+                              "or customize them, use the smart_quotes option in "
+                              "docutils.conf.",
+                              RemovedInSphinx17Warning)
+                self.settings['smart_quotes'] = self.config.html_use_smartypants
+            for tag in normalize_language_tag(language):
+                if tag in smartchars.quotes:
+                    break
+            else:
+                self.settings['smart_quotes'] = False
```

Comment:
@tk0miya this may need some revision: if ``'smart_quotes'`` was in ``self.settings`` and set to True then the for loop and its break and else clauses should arguably be executed, as they check if language supports smart quotes. Currently they are not executed if builder added to environment a ``'smart_quotes'`` True setting

Also possibly this for loop could be occasion to check case that language is CJK language and then use False as default setting, not True. Relates #4142

Reply:
Thanks, I missed that. Fixed at e03ec5b.